app/javascript/lib/validation/shared-validation.js

```diff
@@ -186,3 +188,23 @@ export function validateGroupColumnCounts(headers, line, isLastLine, dataObj) {
   }
   return issues
 }
+
+/**
+ * Timeout the CSFV if taking longer than 10 seconds
+ *
+ */
+export function timeOutCSFV(chunker) {
+  const maxTime = 10000 // in milliseconds this equates to 10 seconds
+  const maxRealTime = chunker.startTime + maxTime
+  const currentTime = new Date().getTime()
```

Comment:
Blocker: [roughly 37% faster](https://jsbench.me/qqkzg2wy98/1), and functionally equivalent for this use case:

```suggestion
  const currentTime = Date.now()
```